TidepoolOnboardingKitUI/Views/OnboardingSectionSheetButton.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  OnboardingSectionSheetButton.swift
+//  TidepoolOnboardingKitUI
+//
+//  Created by Darin Krauss on 3/10/21.
+//  Copyright © 2021 Tidepool Project. All rights reserved.
+//
+
+import SwiftUI
+import LoopKitUI
+
+struct OnboardingSectionSheetButton<Destination: View, Content: View>: View {
+    @EnvironmentObject var onboardingViewModel: OnboardingViewModel
+    @EnvironmentObject var displayGlucoseUnitObservable: DisplayGlucoseUnitObservable
+    @Environment(\.colorPalette) var colorPalette: LoopUIColorPalette
+
+    @State private var isActive: Bool = false
+
+    private let section: OnboardingSection
+    private let destination: Destination
+    private let content: Content
+
+    init(section: OnboardingSection, destination: Destination, @ViewBuilder content: () -> Content) {
+        self.section = section
+        self.destination = destination
+        self.content = content()
+    }
+
+    var body: some View {
+        Button(action: { isActive = true }) {
+            content
+        }
+        .sheet(isPresented: $isActive) {
+            destination
+                .environmentObject(onboardingViewModel)
+                .environmentObject(displayGlucoseUnitObservable)
+                .environment(\.colorPalette, colorPalette)
+                .environment(\.dismiss, {
+                    isActive = false
```

Comment:
After you brought this up at dev sync, I encountered the same issue of needing to include environment items in the modal environment. This must be how SwiftUI works, eh?

Reply:
Yes, I think so. It looks there was a fix included in an iOS beta, but it appears to have been yanked at the last minute and was not released. I can see a number of issues of blindly forwarding the entire environment to a modal/sheet.

Reply:
Yeah but it is too bad they don't have a mnemonic for "just copy all the environment, so I don't have to have code duplication".